packages/utils/src/SparseMerkleTree.ts

```diff
@@ -109,27 +126,30 @@ export class SparseMerkleTree {
      * @param leafKey A key of an existing or a non-existing entry.
      * @returns A merkle proof of a given `leafKey`
      */
-    public createProof(leafKey: bigint): bigint[] {
+    public createProof(leafKey: bigint): bigint[][] {
         assert(
             leafKey < this.numLeaves,
             `leaf key ${leafKey} exceeds total number of leaves ${this.numLeaves}`
         )
 
-        const siblingNodeHashes: bigint[] = []
+        const siblingNodeHashes: bigint[][] = Array(this.height)
+            .fill(null)
+            .map(() => [])
```

Comment:
I think it is the same as `Array(this.height).fill([])`

Reply:
It's not, `Array(this.height).fill([])` uses a single element by reference. It's like doing this:

```js
const a = []
const arr = [a, a, a, a]
```

If any of them get mutated they all get mutated. Need to map into a function that creates a new array for each index.